src/dotnet/Telemetry.cs

```diff
@@ -31,12 +42,26 @@ public Telemetry()
                 return;
             }
 
+            int sampleRate = ReciprocalSampleRateValue;
+            bool isCITestMachine = Env.GetEnvironmentVariableAsBool(ContinousIntegrationFlag);
```

Comment:
I will investigate, but I believe it was for the CI machines that you mentioned.
it has been refactored for now though to be just testMachines.